airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -45,16 +51,34 @@ public class IntegrationRunner {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(IntegrationRunner.class);
 
+  /**
+   * Filters threads that should not be considered when looking for orphaned threads at shutdown of
+   * the integration runner.
+   * <p>
+   * </p>
+   * <b>N.B.</b> Daemon threads don't block the JVM if the main `currentThread` exits, so they are not
+   * problematic. Additionally, ignore database connection pool threads, which stay active so long as
+   * the database connection pool is open.
+   */
+  @VisibleForTesting
+  static final Predicate<Thread> ORPHANED_THREAD_FILTER = runningThread -> !runningThread.getName().equals(Thread.currentThread().getName())
+      && !runningThread.isDaemon()
+      && !runningThread.getName().startsWith("pool-")
```

Comment:
I think we can get rid of the thread name filters now that we are sure that the exit hook will only be called in the finally block at the end of the sync.

Reply:
@subodh1810 I will test out removing the name filters locally and see what happens.

Reply:
@subodh1810 Removing the thread name checks appears to work in the concurrent case and the existing tests using the serial case still pass, so I have pushed up the suggested change.